File: baekjoon/2477/2477.py
'''
참외밭 python 풀이
https://doctcoder.tistory.com/entry/%EC%B0%B8%EC%99%B8%EB%B0%AD-%ED%92%80%EC%9D%B4
'''
k=int(input())
total=[]
x=[]
y=[]
small=[]
for i in range(6):
	dir,leng=map(int,input().split())
	if dir==3 or dir==4: x.append(leng)
	if dir==1 or dir==2: y.append(leng)
	total.append([dir,leng])
for i in range(6):
	if total[i][0]==total[(i+2)%6][0]:
		small.append(total[(i+1)%6][1])
print((max(x)*max(y)-small[0]*small[1])*k)

File: baekjoon/2477/2477.cpp
/*
참외밭 C++ 풀이
https://doctcoder.tistory.com/entry/%EC%B0%B8%EC%99%B8%EB%B0%AD-%ED%92%80%EC%9D%B4
*/
#include <iostream>

using namespace std;

int main()
{
	pair<int,int> cor[12];
	int k,a,b,ba,sa;
	cin>>k;
	for(int i=0;i<6;i++)
	{
		cin>>a>>b;
		cor[i]=cor[i+6]={a,b};
	}
	for(int i=0;i<12;i++)
	{
		if(cor[i].first==cor[i-2].first&&cor[i-1].first==cor[i-3].first)
		{
			ba=cor[i+1].second*cor[i+2].second;
			sa=cor[i-1].second*cor[i-2].second;
			break;
		}
	}
	cout<<k*(ba-sa);
}
